### PR TITLE
ISPN-10608 embedded docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-10608 

Needs backport to 10.1.x

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/proc_running_embedded.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+...
+
+      GlobalConfigurationBuilder global = GlobalConfigurationBuilder.defaultClusteredBuilder();
+            DefaultCacheManager cacheManager = new DefaultCacheManager(global.build());
+            ConfigurationBuilder builder = new ConfigurationBuilder();
+            builder.clustering().cacheMode(CacheMode.DIST_SYNC);
+      cacheManager.administration().getOrCreate("myCache", builder.build());
+----
+
+The preceding code initializes a default, local Cache Manager. Cache Managers
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: Incorrect: the above code creates a clustered cache manager.

Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: argh. sorry that should be totally obvious. I guess I got hold of a bad batch of TP moonshine. fixed.

File: documentation/src/main/asciidoc/topics/dependencies_maven/ispn_embedded.xml [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
-  <artifactId>infinispan-embedded</artifactId>
-  <version>${version.infinispan}</version>
-</dependency>
+<dependencyManagement>
+    <dependencies>
+       <dependency>
+           <groupId>org.infinispan</groupId>
+           <artifactId>infinispan-bom</artifactId>
+           <version>${version.infinispan}</version>
+       </dependency>
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/dependencies_maven/ispn_embedded.xml]: you don't have to use the bom, but if you do, you need to import it:
```suggestion
       <dependency>
           <groupId>org.infinispan</groupId>
           <artifactId>infinispan-bom</artifactId>
           <version>${version.infinispan}</version>
           <type>pom</type>
           <scope>import</scope>
       </dependency>
```
you have an example in spring-boot repo: https://github.com/infinispan/infinispan-spring-boot/blob/master/pom.xml#L95

File: documentation/src/main/asciidoc/topics/proc_running_embedded.adoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+
+...
+
+      GlobalConfigurationBuilder global = GlobalConfigurationBuilder.defaultClusteredBuilder();
+            DefaultCacheManager cacheManager = new DefaultCacheManager(global.build());
+            ConfigurationBuilder builder = new ConfigurationBuilder();
+            builder.clustering().cacheMode(CacheMode.DIST_SYNC);
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: can you align the spaces?

File: documentation/src/main/asciidoc/topics/proc_running_embedded.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+----
+
+The preceding code initializes a default, clustered Cache Manager. Cache
+Managers contain your cache definitions and control cache lifecycles.
+
+{brandname} does not provide default cache definitions so after initializing
+the default Cache Manager, you need to add at least one cache instance. This
+example uses the `ConfigurationBuilder` class to create a cache definition that
+uses the distributed, synchronous cache mode. You then call the `getOrCreate()`
+method that either creates a cache named "myCache" or returns it if it already
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: is it worth to mention that the cache is created in all the nodes in the cluster?

File: documentation/src/main/asciidoc/topics/proc_running_embedded.adoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+= Running {brandname} as an Embedded Library
+Learn how to run {brandname} as an embedded data store in your project.
+
+.Procedure
+
+* Initialize the default Cache Manager and add a cache definition as follows:
+
+[source,java,options="nowrap"]
+----
+import org.infinispan.Cache;
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: what do you think of keeping only the code and remove the imports?

File: documentation/src/main/asciidoc/topics/proc_running_embedded.adoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+.Reference
+
+* link:{config_docs}#programmatic-configuring[Configuring {brandname} Programmatically]
+* link:{javadocroot}/org/infinispan/Cache.html[org.infinispan.Cache]
+* link:{javadocroot}/org/infinispan/commons/api/CacheContainerAdmin.html[org.infinispan.commons.api.CacheContainerAdmin]
+* link:{javadocroot}/org/infinispan/configuration/cache/CacheMode.html[org.infinispan.configuration.cache.CacheMode]
+* link:{javadocroot}/org/infinispan/configuration/cache/Configuration.html[org.infinispan.configuration.cache.Configuration]
+* link:{javadocroot}/org/infinispan/configuration/cache/ConfigurationBuilder.html[org.infinispan.configuration.cache.ConfigurationBuilder]
+* link:{javadocroot}/org/infinispan/configuration/global/GlobalConfigurationBuilder.html[org.infinispan.configuration.global.GlobalConfigurationBuilder]
+* link:{javadocroot}org/infinispan/manager/DefaultCacheManager.html[org.infinispan.manager.DefaultCacheManager]
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: A `/` is missing.
```suggestion
* link:{javadocroot}/org/infinispan/manager/DefaultCacheManager.html[org.infinispan.manager.DefaultCacheManager]
```

File: documentation/src/main/asciidoc/topics/proc_running_embedded.adoc [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+the default Cache Manager, you need to add at least one cache instance. This
+example uses the `ConfigurationBuilder` class to create a cache definition that
+uses the distributed, synchronous cache mode. You then call the `getOrCreate()`
+method that either creates a cache named "myCache" or returns it if it already
+exists.
+
+.Next steps
+
+Now that you have a running Cache Manager with a cache created, you can add
+some more cache definitions, configure a cluster transport layer, put some data
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/proc_running_embedded.adoc]: the transport must be configured before instantiating the `DefaultCacheManager`.

**Comment by pruivo** [comment]
integrated! thanks @oraNod !

**Pull request merged**